x-pack/plugins/infra/public/alerting/metric_threshold/lib/transform_metrics_explorer_data.ts

```diff
@@ -14,8 +14,8 @@ export const transformMetricsExplorerData = (
 ) => {
   const { criteria } = params;
   if (criteria && data) {
-    const firstSeries = first(data.series) as any;
-    const series = firstSeries.rows.reduce((acc: any, row: any) => {
+    const firstSeries = first(data.series)!;
+    const series = firstSeries.rows.reduce((acc, row) => {
```

Comment:
Isn't the assertion `first(data.series)!` risky without checking `data.series.length`? How about we bind the series to a name early and check it for `undefined` directly?

```typescript
  const { criteria } = params;
  const firstSeries = first(data?.series);
  if (criteria && firstSeries) {
    const series = firstSeries.rows.reduce((acc, row) => {
```